src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1317,6 +1404,22 @@ Literal &Literal::operator-=(const Literal &other) noexcept {
 }
 
 Literal Literal::mul(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
+    if (this->datatype_eq<datatypes::xsd::YearMonthDuration>() && other.datatype_eq<datatypes::xsd::Double>()) {
+        auto ym = this->value<datatypes::xsd::YearMonthDuration>();
+        auto d = other.value<datatypes::xsd::Double>();
+        auto r = std::round(static_cast<double>(ym.count()) * d);
+        if (!datatypes::registry::util::fits_into<int64_t>(r))
+            return Literal{};
+        return make_typed_from_value<datatypes::xsd::YearMonthDuration>(std::chrono::months{static_cast<int64_t>(r)});
+    } else if (this->datatype_eq<datatypes::xsd::DayTimeDuration>() && other.datatype_eq<datatypes::xsd::Double>()) {
```

Comment:
According to XPath: No.
(It does not even specify a function for DateTime + Duration, but defined an algorythm to do the calculation.
I added it anyway, because it needs less type checks.)
(Maybe because Duration is a DayTimeDuration + a YearMonthDuration internaly anyway.)